sphinx/directives/other.py

```diff
@@ -124,7 +124,7 @@ def parse_content(self, toctree: addnodes.toctree) -> List[Node]:
                         break
                 # absolutize filenames
                 docname = docname_join(self.env.docname, docname)
-                if url_re.match(ref) or ref == 'self':
+                if url_re.match(ref) or ref in ('self', 'genindex', 'modindex', 'search'):
```

Comment:
I'd prefer if this 4-tuple was factored out somewhere as it is used more than once, but there aren't any great locations that come to mind.